### PR TITLE
Bug 1343355 - Make sure a normal tab exists when doing a session restore.

### PR DESCRIPTION
**Pull request by farhanpatel**

### CODE REVIEW

File: Client/Frontend/Browser/TabManager.swift [hunk 1]
```diff
@@ -740,6 +740,10 @@ extension TabManager {
             let tab = addTab()
             selectTab(tab)
         }
+        // Always make sure there is a single normal tab.
+        if normalTabs.isEmpty {
+            addTab()
+        }
```
Comment (thebnich) [inline Client/Frontend/Browser/TabManager.swift]: The four lines above this (with the `count == 0`) are redundant now, so I'd remove them. You'll probably want to change your code to something like this:

```
if normalTabs.isEmpty {
    let tab = addTab()
    if selectedTab == nil {
        selectTab(tab)
    }
}
```

Comment (bkmunar) [inline Client/Frontend/Browser/TabManager.swift]: Better than that, can't we just use `addTabAndSelect()`?

Comment (thebnich) [inline Client/Frontend/Browser/TabManager.swift]: No, because if we previously had a private tab selected, we don't want to lose that selection.

Comment (bkmunar) [inline Client/Frontend/Browser/TabManager.swift]: Oh ok cool

**Pull request merged**